UnityProject/Assets/Scripts/Editor/Inspector/SubscriptionControllerEditor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using Electric.Inheritance;
+using UnityEngine;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine.SceneManagement;
+
+[CustomEditor(typeof(SubscriptionController), true)]
+public class SubscriptionControllerEditor : Editor
+{
+	private SubscriptionController controller;
+	private bool isSelecting;
+
+	void OnEnable()
+	{
+		SceneView.duringSceneGui += OnScene;
+	}
+
+	void OnDisable()
+	{
+		SceneView.duringSceneGui -= OnScene;
+	}
+
+	public override void OnInspectorGUI()
+	{
+		base.OnInspectorGUI();
+
+		if (!isSelecting)
+		{
+			if (GUILayout.Button("Begin Selecting"))
+			{
+				isSelecting = true;
+				controller = (SubscriptionController)target;
+			}
+		}
+		else
+		{
+			if (GUILayout.Button("Stop Selecting"))
+			{
+				isSelecting = false;
+				controller = null;
+			}
+		}
+	}
+
+	void OnScene(SceneView scene)
+	{
+		//skip if not selecting
+		if (!isSelecting || controller == null)
+			return;
+
+		Event e = Event.current;
+		if (e == null)
+			return;
+
+		if (HasPressedEscapeKey(e))
+		{
+			isSelecting = false;
+			return;
+		}
+
+		if (HasPressedLeftClick(e))
+		{
+			Ray ray = HandleUtility.GUIPointToWorldRay(e.mousePosition);
+			RaycastHit2D[] hits = Physics2D.RaycastAll(ray.origin, ray.direction);
+			List<GameObject> hitObjects = new List<GameObject>();
```

Comment:
hey, I did some changes to this because it was having trouble picking up objects, sometimes doubling them so I cast them to game objects then put them an hashset to remove duplicates